File: backend/app/schemas/notification_settings.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class NotificationSettingsResponse(BaseModel):
    id: int
    telegram_enabled: bool
    telegram_chat_id: Optional[str]
    browser_push_enabled: bool
    whatsapp_enabled: bool
    whatsapp_phone: Optional[str]
    created_at: datetime
    updated_at: Optional[datetime]

    model_config = {"from_attributes": True}


class NotificationSettingsUpdate(BaseModel):
    telegram_enabled: Optional[bool] = None
    telegram_chat_id: Optional[str] = None
    browser_push_enabled: Optional[bool] = None
    whatsapp_enabled: Optional[bool] = None
    whatsapp_phone: Optional[str] = None


class TestTelegramRequest(BaseModel):
    chat_id: str = Field(..., min_length=1, max_length=64)


class TestTelegramResponse(BaseModel):
    success: bool
    message: str


class TestWhatsAppRequest(BaseModel):
    phone: str = Field(..., min_length=7, max_length=32)


class TestWhatsAppResponse(BaseModel):
    success: bool
    message: str

File: frontend/types/notificationSettings.ts
export interface NotificationSettings {
  id: number
  telegram_enabled: boolean
  telegram_chat_id: string | null
  browser_push_enabled: boolean
  whatsapp_enabled: boolean
  whatsapp_phone: string | null
  created_at: string
  updated_at: string | null
}

export interface NotificationSettingsUpdate {
  telegram_enabled?: boolean
  telegram_chat_id?: string | null
  browser_push_enabled?: boolean
  whatsapp_enabled?: boolean
  whatsapp_phone?: string | null
}

export interface TestTelegramRequest {
  chat_id: string
}

export interface TestTelegramResponse {
  success: boolean
  message: string
}

export interface TestWhatsAppRequest {
  phone: string
}

export interface TestWhatsAppResponse {
  success: boolean
  message: string
}
